File: app/android/app/src/main/java/com/gonezo/multiplatform/audioextraction/application/SourceLoader.java
package com.gonezo.multiplatform.audioextraction.application;

import com.gonezo.multiplatform.audioextraction.contract.ExtractionRequest;
import com.gonezo.multiplatform.audioextraction.domain.model.SourceAudio;

public interface SourceLoader {
  SourceAudio load(ExtractionRequest request);
}

File: core/bin/main/com/gonezo/audioextraction/application/pipeline/SourceLoader.kt
package com.gonezo.audioextraction.application.pipeline

import com.gonezo.audioextraction.domain.contract.ExtractionRequest
import com.gonezo.audioextraction.domain.model.SourceAudio

interface SourceLoader {
  fun load(request: ExtractionRequest): SourceAudio
}
